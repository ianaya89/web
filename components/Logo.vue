<template>
  <nuxt-link :to="localePath({name:'index'})" class="logo--so" data-cy="logo-link">
    <img :src="logoSource" :class="logoClasses" alt="SPONSOR.ONLINE">
  </nuxt-link>
</template>

<script>
export default {
  props: {
    variant: {
      type: String,
      default: 'dark',
      validator: value => {
        return ['dark', 'light', 'mobile'].indexOf(value) !== -1
      }
    }
  },

  computed: {
    logoSource() {
      if (this.variant === 'light') {
        return require('~/assets/images/logos/so-logo-light.svg')
      } else if (this.variant === 'dark') {
        return require('~/assets/images/logos/so-logo-dark.svg')
      } else {
        return require('~/assets/images/logos/so-symbol-primary.svg')
      }
    },

    logoClasses() {
      if (this.variant === 'mobile') {
        return 'w-10 h-10'
      } else {
        return 'w-32'
      }
    }
  }
}
</script>

<style lang="postcss" scoped>
.logo--so {
  @apply .flex;
}
</style>
