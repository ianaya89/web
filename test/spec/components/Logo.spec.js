import { shallowMount } from '@vue/test-utils'
import Logo from '@/components/Logo'

describe('Logo', () => {
  it('renders default logo', () => {
    const wrapper = shallowMount(Logo, {
      propsData: {
        variant: 'dark'
      }
    })
    expect(wrapper.element).toMatchSnapshot()
    wrapper.setProps({
      variant: 'light'
    })
    expect(wrapper.element).toMatchSnapshot()
    wrapper.setProps({
      variant: 'mobile'
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
