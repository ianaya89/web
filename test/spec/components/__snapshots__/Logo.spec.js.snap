// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo renders default logo 1`] = `
<a
  class="logo--so"
  data-cy="logo-link"
>
  <img
    alt="SPONSOR.ONLINE"
    class="w-32"
    src="\\"so-logo-dark.svg\\""
  />
</a>
`;

exports[`Logo renders default logo 2`] = `
<a
  class="logo--so"
  data-cy="logo-link"
>
  <img
    alt="SPONSOR.ONLINE"
    class="w-32"
    src="\\"so-logo-dark.svg\\""
  />
</a>
`;

exports[`Logo renders default logo 3`] = `
<a
  class="logo--so"
  data-cy="logo-link"
>
  <img
    alt="SPONSOR.ONLINE"
    class="w-32"
    src="\\"so-logo-dark.svg\\""
  />
</a>
`;
